### PR TITLE
Added switch for switching between dialog and integrated view visitor…

### PR DESCRIPTION
**Pull request by Pelkar**
… code

MOB-2072

Added a way to switch between a dialog and integrated view for Visitor Code
switch

### CODE REVIEW

File: app/src/main/res/layout/main_fragment.xml [hunk 1]
```diff
@@ -184,6 +184,46 @@
                 app:layout_constraintStart_toStartOf="@+id/start_guideline"
                 app:layout_constraintTop_toBottomOf="@+id/clear_session_button" />
 
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:id="@+id/visitor_code_switch_container"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                app:layout_constraintTop_toBottomOf="@id/visitor_code_button"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent" >
```
Comment (gugalo) [inline app/src/main/res/layout/main_fragment.xml:193]: NIT: would try to move this container a bit closer to the Visitor Code button. Maybe using negative margin or something else

**Review by gugalo — APPROVED** [review]

File: app/src/main/res/layout/main_fragment.xml (hunk 1 above)
Comment (DavDo) [inline app/src/main/res/layout/main_fragment.xml:193]: NIT: `ConstraintLayout` provides all necessary mechanisms to not nest `ConstraintLayout`'s to each other. So this container is redundant. You can constraint views inside parent `ConstraintLayout` and use the `Group` widget to work with view groups of views e.g. change visibility etc.

**Review by DavDo — APPROVED** [review]

**Comment by andrews-moc** [comment]
I don't have any comments, the implementation looks good.
However, my concern is that the Main screen might become overloaded soon.
@Pelkar , @gugalo , @DavDo , @AndriiHorishniiMOC What do you think regarding moving this setting to the Settings screen?

**Comment by DavDo** [comment]
> I don't have any comments, the implementation looks good. However, my concern is that the Main screen might become overloaded soon. @Pelkar , @gugalo , @DavDo , @AndriiHorishniiMOC What do you think regarding moving this setting to the Settings screen?

Agree.

**Comment by Pelkar** [comment]
> I don't have any comments, the implementation looks good. However, my concern is that the Main screen might become overloaded soon. @Pelkar , @gugalo , @DavDo , @AndriiHorishniiMOC What do you think regarding moving this setting to the Settings screen?

To me, it would look out of place, especially with the current implementation. What I mean by that is that Settings screen is a place for permanent changes, this however is not stored anywhere and will reset with relaunch of the application.

So either we would need to make the selection persist or keep it in the main fragment.

Essentially it is the teams choice

**Comment by DavDo** [comment]
> > I don't have any comments, the implementation looks good. However, my concern is that the Main screen might become overloaded soon. @Pelkar , @gugalo , @DavDo , @AndriiHorishniiMOC What do you think regarding moving this setting to the Settings screen?
> 
> To me, it would look out of place, especially with the current implementation. What I mean by that is that Settings screen is a place for permanent changes, this however is not stored anywhere and will reset with relaunch of the application.
> 
> So either we would need to make the selection persist or keep it in the main fragment.
> 
> Essentially it is the teams choice

Sounds reasonable to me. Maybe in this case it makes sense to let it as it is and later, try to redesign/refactor this screen(e.g move the end engagement button to appBar, place similar items next to each other, move some rarely used items(clear session) to menu). It will also lead to changes in instrumentation tests.

**Comment by Pelkar** [comment]
If @andrews-moc is ok with it, I would go ahead with the merge. We can do the redesign at some point in the future where we can rethink the whole view from the ground up if need be.

**Comment by andrews-moc** [comment]
Okay, agree ✅

**Pull request merged**